tensorboard/components/vz_chart_helpers/plottable-interactions.js

```diff
@@ -0,0 +1,144 @@
+var vz_chart_helpers;
+(function(vz_chart_helpers) {
+  // HACK: parentElement does not work for webcomponents.
+  function getHtmlElementAncestors(elem) {
+    const elems = [];
+    while (elem && elem instanceof HTMLElement) {
+      elems.push(elem);
+      if (elem.assignedSlot) {
+        elem = elem.assignedSlot;
+      } else if (!elem.parentElement) {
+        const parentNode = elem.parentNode;
+        if (parentNode instanceof DocumentFragment) {
+          elem = parentNode.host;
+        } else {
+          // <html>.parentNode == <html>
+          elem = parentNode !== elem ? parentNode : null;
+        }
+      } else {
+        elem = elem.parentElement;
+      }
+    }
+    return elems;
+  }
+
+  const _IDENTITY_TRANSFORM = [1, 0, 0, 1, 0, 0];
+
+  // Forked from https://github.com/palantir/plottable/blob/b6e36fbd4d8d7cba579d853b9f35cc260d1243bf/src/utils/mathUtils.ts#L173-L202
+  // The only difference is the implementation of the getHtmlElementAncestors.
+  function getCumulativeTransform(element) {
+    const elems = getHtmlElementAncestors(element);
+
+    let transform = _IDENTITY_TRANSFORM;
+    let offsetParent = null;
+    for (const elem of elems) {
+      // apply css transform from any ancestor element
+      const elementTransform = Plottable.Utils.DOM.getElementTransform(elem);
+      if (elementTransform != null) {
+        const midX = elem.clientWidth / 2;
+        const midY = elem.clientHeight / 2;
+        transform = Plottable.Utils.Math.multiplyTranslate(transform, [midX, midY]);
+        transform = Plottable.Utils.Math.multiplyMatrix(transform, Plottable.Utils.Math.invertMatrix(elementTransform));
+        transform = Plottable.Utils.Math.multiplyTranslate(transform, [-midX, -midY]);
+      }
+
+      // apply scroll offsets from any ancestor element
+      let offsetX = elem.scrollLeft;
+      let offsetY = elem.scrollTop;
+
+      // apply client+offset from only acenstor "offsetParent"
+      if (offsetParent === null || elem === offsetParent) {
+        offsetX -= elem.offsetLeft + elem.clientLeft;
+        offsetY -= elem.offsetTop + elem.clientTop;
+        offsetParent = elem.offsetParent;
+      }
+      transform = Plottable.Utils.Math.multiplyTranslate(transform, [offsetX, offsetY]);
+    }
+    return transform;
+  }
+
+  class CustomTranslator extends Plottable.Utils.Translator {
+    computePosition(clientX, clientY) {
+      const clientPosition = {
+        x: clientX,
+        y: clientY,
+      };
+
+      const transform = getCumulativeTransform(this._rootElement);
+      if (transform == null) {
+        return clientPosition;
+      }
+
+      const transformed = Plottable.Utils.Math.applyTransform(transform, clientPosition);
+      return transformed;
+    }
+  }
+
+  class MouseDispatcher extends Plottable.Dispatchers.Mouse {
+    constructor(component) {
+      super(component);
+      // eventTarget is `document` by default. Change it to the root of chart.
```

Comment:
From your comment on https://github.com/palantir/plottable/issues/3350#issuecomment-460486874 it sounded like access to _eventTarget was all that was needed, but it looks like the custom translator is also necessary?  Am I understanding that correctly?

Reply:
Yup correct. In order for events to correctly trigger, I needed the `eventTarget` change but soon found out, when I was actually fixing things, that mouse event translation (Plottable changes pointer coordinate to relative to the SVG) was very incorrect.